File: Day2_ControlFlow_Go/easy/easy_01.go
// FILE: easy_01.go
// TOPIC: Control Flow Deep Dive - Day 2 (Go)
// DIFFICULTY: Easy
//
// TASK: Short-Circuit Evaluation & Guard Clause with if/else
// -----------------------------------------------------------
// In Go, logical AND (&&) stops immediately if the left condition is false.
// Logical OR (||) stops immediately if the left condition is true.
//
// This prevents nil pointer panics!
// For example:
//   if ptr != nil && *ptr > 50  <-- *ptr is NEVER dereferenced if ptr is nil!
//
// 1. Declare a *int pointer `var ptr *int` (zero value is nil).
// 2. Write an if-statement checking:
//      if ptr != nil && *ptr > 50:
//          print "Valid and greater than 50"
//      else:
//          print "Safety triggered: pointer is nil or value <= 50"
// 3. Verify that running this does NOT panic.
// 4. Next, assign `val := 85` and `ptr = &val`.
// 5. Run the same check again.
//
// EXPECTED OUTPUT:
//   Safety triggered: pointer is nil or value <= 50
//   Valid and greater than 50

package main

import "fmt"

func main() {
	// TODO: Write your solution here.

}

File: Day2_ControlFlow/easy/easy_01.cpp
/*
 * FILE: easy_01.cpp
 * TOPIC: Control Flow Deep Dive - Day 2
 * DIFFICULTY: Easy
 *
 * TASK: Short-Circuit Evaluation & Guard Clause with if/else
 * -----------------------------------------------------------
 * In C++, logical AND (&&) stops immediately if the left condition is FALSE.
 * Logical OR (||) stops immediately if the left condition is TRUE.
 *
 * This prevents crashes! For example:
 *   if (ptr != nullptr && *ptr > 10)  <-- *ptr is NEVER dereferenced if ptr is null!
 *
 * 1. Declare an int pointer `ptr = nullptr`.
 * 2. Write an if-statement that checks:
 *      if `ptr` is not null AND the value at `ptr` is greater than 50:
 *          print "Valid and greater than 50"
 *      else:
 *          print "Safety triggered: pointer is null or value <= 50"
 * 3. Verify that running this does NOT crash your program.
 * 4. Next, assign `ptr = &val` where `int val = 85`.
 * 5. Run the same check again.
 *
 * EXPECTED OUTPUT:
 *   Safety triggered: pointer is null or value <= 50
 *   Valid and greater than 50
 */

#include <iostream>
using namespace std;

int main() {
    // TODO: Write your solution here.



    return 0;
}
